### PR TITLE
feat(manager): runtime version

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Add new manager for runtime versions.

Initial focus is on python runtimes used by
- https://devcenter.heroku.com/articles/python-runtimes
- https://www.koyeb.com/docs/build-and-deploy/build-from-git/python
- https://docs.cloudfoundry.org/buildpacks/python/index.html
- etc

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Example repository: https://github.com/setchy/renovate-heroku-runtime/pull/3

### CODE REVIEW

File: lib/modules/manager/heroku-runtime/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { DockerDatasource } from '../../datasource/docker';
+import * as dockerVersioning from '../../versioning/docker';
+
+export { extractPackageFile } from './extract';
+
+export const supportedDatasources = [DockerDatasource.id];
+
+export const defaultConfig = {
+  fileMatch: ['^runtime.txt$'],
+  versioning: dockerVersioning.id,
```
Comment (viceice) [inline lib/modules/manager/heroku-runtime/index.ts:11]:
```suggestion
```
default for docker datasource

File: lib/modules/manager/heroku-runtime/index.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { Category } from '../../../constants';
+import { DockerDatasource } from '../../datasource/docker';
+import * as dockerVersioning from '../../versioning/docker';
+
+export { extractPackageFile } from './extract';
+
+export const supportedDatasources = [DockerDatasource.id];
```
Comment (viceice) [inline lib/modules/manager/heroku-runtime/index.ts:7]: please export some urls as reference links

File: lib/modules/manager/heroku-runtime/extract.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { DockerDatasource } from '../../datasource/docker';
+import type { PackageDependency, PackageFileContent } from '../types';
+
+export function extractPackageFile(content: string): PackageFileContent {
+  const dep: PackageDependency = {
+    depName: 'python',
+    commitMessageTopic: 'Python',
+    currentValue: content.replaceAll('python-', '').trim(),
```
Comment (viceice) [inline lib/modules/manager/heroku-runtime/extract.ts:8]: is there only one line in this file? can there be other values? we should validate and document that only python is yet supported.

maybe also add a preview state note.

Comment (setchy) [reply lib/modules/manager/heroku-runtime/extract.ts]: changed to use a regex match group

**Review by secustor — COMMENTED** [review]
Not sure about the name as this file and content seemed to be used by other systems too. 

https://www.koyeb.com/docs/build-and-deploy/build-from-git/python
https://docs.cloudfoundry.org/buildpacks/python/index.html

Is this maybe based on a common system?

**Comment by setchy** [comment]
How interesting @secustor - I didn't realize that.

So perhaps this is better off as a `python-runtime` manager

**Comment by rarkins** [comment]
Can we make it a "runtime" manager more generically? If the python version always has a prefix, then in future there might be others, e.g. ruby-1.2.3 and we can then immediately accommodate them without a rename.

File: lib/modules/manager/runtime-version/extract.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+);
+
+export function extractPackageFile(content: string): PackageFileContent | null {
+  const regexResult = pythonRuntimeRegex.exec(content);
+  const runtimeVersion = regexResult?.groups?.version;
+
+  if (runtimeVersion) {
+    const dep: PackageDependency = {
+      depName: 'python',
+      commitMessageTopic: 'Runtime Version',
```
Comment (secustor) [inline lib/modules/manager/runtime-version/extract.ts:16]: Why the specific message topic?

File: lib/modules/manager/runtime-version/index.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import type { Category } from '../../../constants';
+import { DockerDatasource } from '../../datasource/docker';
+
+export { extractPackageFile } from './extract';
+
+export const displayName = 'Runtime Version';
+
+export const defaultConfig = {
+  fileMatch: ['^runtime.txt$'],
```
Comment (secustor) [inline lib/modules/manager/runtime-version/index.ts:9]:
```suggestion
  fileMatch: ['(^|/)runtime.txt$'],
```
Can this really be only at the root of repos?

File: lib/modules/manager/runtime-version/extract.ts (hunk 4 above)
Comment (setchy) [reply lib/modules/manager/runtime-version/extract.ts:16]: Copied the structure from one of the existing managers.  Easily removed

File: lib/modules/manager/runtime-version/index.ts (hunk 5 above)
Comment (setchy) [reply lib/modules/manager/runtime-version/index.ts:9]: Heroku
> To specify a Python runtime, add a runtime.txt file to your app’s root directory that declares the exact version number to use.

Koyeb
> To specify the Python version to use, add a runtime.txt file in the root of your project directory with one of the versions mentioned above

CloudFoundry
> You can specify a version of the Python runtime by including it within a runtime.txt file. For example:

Comment (setchy) [reply lib/modules/manager/runtime-version/index.ts:9]: I'm not fussed if we relax this though

**Review by secustor — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**